webapi/src/test/scala/org/knora/webapi/e2e/v2/StandoffRouteV2R2RSpec.scala

```diff
@@ -0,0 +1,151 @@
+
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.{Await, ExecutionContextExecutor}
+
+
+/**
+  * End-to-end test specification for the search endpoint. This specification uses the Spray Testkit as documented
+  * here: http://spray.io/documentation/1.2.2/spray-testkit/
+  */
+class StandoffRouteV2R2RSpec extends R2RSpec {
+
+    override def testConfigSource: String =
+        """
+          |# akka.loglevel = "DEBUG"
+          |# akka.stdout-loglevel = "DEBUG"
+        """.stripMargin
+
+    private val responderManager = system.actorOf(Props(new ResponderManager with LiveActorMaker), name = RESPONDER_MANAGER_ACTOR_NAME)
+    private val storeManager = system.actorOf(Props(new StoreManager with LiveActorMaker), name = STORE_MANAGER_ACTOR_NAME)
+
+    private val standoffPath = StandoffRouteV2.knoraApiPath(system, settings, log)
+
+    implicit private val timeout: Timeout = settings.defaultRestoreTimeout
+
+    implicit def default(implicit system: ActorSystem): RouteTestTimeout = RouteTestTimeout(new DurationInt(15).second)
+
+    implicit val ec: ExecutionContextExecutor = system.dispatcher
+
+    private val anythingUser = SharedTestDataADM.anythingUser1
+    private val anythingUserEmail = anythingUser.email
+
+    private val password = "test"
+
+    object RequestParams {
+
+        val pathToLetterMapping = "_test_data/test_route/texts/mappingForLetter.xml"
+
+        val pathToLetterXML = "_test_data/test_route/texts/letter.xml"
+
+        val pathToLetter2XML = "_test_data/test_route/texts/letter2.xml"
+
+        val pathToLetter3XML = "_test_data/test_route/texts/letter3.xml"
+
+        // Standard HTML is the html code that can be translated into Standoff markup with the OntologyConstants.KnoraBase.StandardMapping
+        val pathToStandardHTML = "_test_data/test_route/texts/StandardHTML.xml"
+
+        val pathToHTMLMapping = "_test_data/test_route/texts/mappingForHTML.xml"
+
+        val pathToHTML = "_test_data/test_route/texts/HTML.xml"
+
+    }
+
+    private val rdfDataObjects = List(
+
+        RdfDataObject(path = "_test_data/all_data/incunabula-data.ttl", name = "http://www.knora.org/data/incunabula"),
+        RdfDataObject(path = "_test_data/demo_data/images-demo-data.ttl", name = "http://www.knora.org/data/00FF/images"),
+        RdfDataObject(path = "_test_data/all_data/anything-data.ttl", name = "http://www.knora.org/data/anything")
+
+    )
+
+    "Load test data" in {
+        Await.result(storeManager ? ResetTriplestoreContent(rdfDataObjects), 360.seconds)
+        Await.result(responderManager ? LoadOntologiesRequestV2(KnoraSystemInstances.Users.SystemUser), 30.seconds)
+    }
+
+    "The Standoff v2 Endpoint" should {
+
+        "create a mapping from a XML" in {
+
+            val xmlFileToSend = new File(RequestParams.pathToLetterMapping)
+
+            val mappingParams =
+                s"""
+                   |{
+                   |    "knora-api:mappingHasName": "LetterMapping",
```

Comment:
@benjamingeer I guess I am supposed to add that to `webapi/src/main/scala/org/knora/webapi/messages/v2/responder/ontologymessages/KnoraApiV2WithValueObjects.scala`, right?

Reply:
Yes, I think that any properties used in requests should be included in `knora-api` as documentation.